### PR TITLE
MERC-6221 Enable full symbol overrides for TP EA

### PR DESCRIPTION
**Pull request by martin-cll**
https://smartcontract-it.atlassian.net/browse/MERC-6221

Enable full symbol overrides for the TP EA. For example, sending the request `{"data":{"endpoint":"forex","from":"EGP","to":"JPY","overrides":{"tp":{"EGP":"FXSPTEGPJPYSPT:GBL.BIL.QTE.RTM!TP"}}}}` would return the exact price for the symbol `FXSPTEGPJPYSPT:GBL.BIL.QTE.RTM!TP` rather than any other prices cached for the base `EPG`. This is relevant in cases where there are multiple instruments (e.g. different expiring forwards) for the same base/quote pair.

This is simply implemented by returning two provider results, the default one with the parsed base and an additional one where the base is the full symbol. Note that I refactored the parsing to use string splitting because forward symbols (e.g. `CEFWDXAUUSDSPT06M:LDN.BIL.QTE.RTM!TP`) can have more characters than spot symbols (e.g. `FXSPTUSDAEDSPT:GBL.BIL.QTE.RTM!TP`)

Also added a `commodities` endpoint that's an alias of the existing `prices` / `forex` endpoint.

### CODE REVIEW

File: packages/sources/tp/src/endpoint/price.ts [hunk 1]
```diff
@@ -47,14 +48,14 @@ export const generateInputParams = (
 const tpOptions: GeneratePriceOptions = {
   sourceName: 'tpSource',
   streamName: 'TP',
-}
+} as const
 
 const inputParameters = generateInputParams(tpOptions)
 const wsTransport = generateTransport(tpOptions)
 
-export const priceEndpoint = new ForexPriceEndpoint({
```
Comment (karen-stepanyan) [inline packages/sources/tp/src/endpoint/price.ts:55]: What's the reason of this change ?

**Review by karen-stepanyan — CHANGES_REQUESTED** [review]
LGTM. Can you add integration test that covers the override part ?

File: packages/sources/tp/src/endpoint/price.ts (hunk 1 above)
Comment (martin-cll) [reply packages/sources/tp/src/endpoint/price.ts:55]: So right now we have the `price` and `forex` aliases with the `ForexPriceEndpoint` class being the same thing as `PriceEndpoint`. Since we want to add new `commodities` alias as well, I thought it would make more sense to use the base `PriceEndpoint` class rather than continue to use the `ForexPriceEndpoint`.

**Comment by martin-cll** [comment]
> LGTM. Can you add integration test that covers the override part ?

Added some integration tests for overrides + WTI mapping.

**Review by karen-stepanyan — APPROVED** [review]
